x-pack/libbeat/common/aws/credentials.go

```diff
@@ -41,13 +43,22 @@ func GetAWSCredentials(config ConfigAWS) (awssdk.Config, error) {
 		return awsConfig, nil
 	}
 
+	// If shared_credential_file is empty, then external.LoadDefaultAWSConfig
+	// function will load AWS config from current user's home directory.
+	// Linux/OSX: "$HOME/.aws/credentials"
+	// Windows:   "%USERPROFILE%\.aws\credentials"
+	if config.SharedCredentialFile != "" {
+		os.Setenv(external.AWSSharedCredentialsFileEnvVar, config.SharedCredentialFile)
+	}
+
 	// If accessKeyID, secretAccessKey or sessionToken is not given, then load from default config
 	// Please see https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html
 	// with more details.
 	if config.ProfileName != "" {
-		return external.LoadDefaultAWSConfig(
+		config, err := external.LoadDefaultAWSConfig(
 			external.WithSharedConfigProfile(config.ProfileName),
```

Comment:
There seems to be a `external.WithSharedConfigFiles()` option, does it set the same setting? could we use something like this instead of overriding environment variables?

Take into account that use of environment variables can be problematic if several metricsets are configured with different values in the same process.

Reply:
Yep! Found one that sets the file: `external.WithSharedConfigFiles`